modular_zubbers/code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -0,0 +1,80 @@
+// Spray tan reagent colors mobs, but not in a way that fucks it up too badly on our mutants.
+/datum/reagent/spraytan/expose_mob(mob/living/exposed_mob, methods=TOUCH, reac_volume, show_message = TRUE)
+	. = ..()
+	if(ishuman(exposed_mob))
+		if(methods & (PATCH|VAPOR))
+			var/mob/living/carbon/human/exposed_human = exposed_mob
+			if(HAS_TRAIT(exposed_human, TRAIT_USES_SKINTONES))
+				switch(exposed_human.skin_tone)
+					if("african1")
+						exposed_human.add_atom_colour(skintone2hex("african2"), WASHABLE_COLOUR_PRIORITY)
+					if("indian")
+						exposed_human.add_atom_colour(skintone2hex("mixed2"), WASHABLE_COLOUR_PRIORITY)
+					if("arab")
+						exposed_human.add_atom_colour(skintone2hex("indian"), WASHABLE_COLOUR_PRIORITY)
+					if("asian2")
+						exposed_human.add_atom_colour(skintone2hex("arab"), WASHABLE_COLOUR_PRIORITY)
+					if("asian1")
+						exposed_human.add_atom_colour(skintone2hex("asian2"), WASHABLE_COLOUR_PRIORITY)
+					if("mediterranean")
+						exposed_human.add_atom_colour(skintone2hex("mixed1"), WASHABLE_COLOUR_PRIORITY)
+					if("latino")
+						exposed_human.add_atom_colour(skintone2hex("mediterranean"), WASHABLE_COLOUR_PRIORITY)
+					if("caucasian3")
+						exposed_human.add_atom_colour(skintone2hex("mediterranean"), WASHABLE_COLOUR_PRIORITY)
+					if("caucasian2")
+						exposed_human.add_atom_colour(skintone2hex("latino"), WASHABLE_COLOUR_PRIORITY)
+					if("caucasian1")
+						exposed_human.add_atom_colour(skintone2hex("caucasian2"), WASHABLE_COLOUR_PRIORITY)
+					if("albino")
+						exposed_human.add_atom_colour(skintone2hex("caucasian1"), WASHABLE_COLOUR_PRIORITY)
+					if("mixed1")
+						exposed_human.add_atom_colour(skintone2hex("mixed2"), WASHABLE_COLOUR_PRIORITY)
+					if("mixed2")
+						exposed_human.add_atom_colour(skintone2hex("mixed3"), WASHABLE_COLOUR_PRIORITY)
+					if("mixed3")
+						exposed_human.add_atom_colour(skintone2hex("african1"), WASHABLE_COLOUR_PRIORITY)
+					if("mixed4")
+						exposed_human.add_atom_colour(skintone2hex("mixed3"), WASHABLE_COLOUR_PRIORITY)
```

Comment:
Whilst this is more efficient I think it'd be much more readable to just have a static keyed lookup list on the spraytan reagent instead. Up to you though